syntax = "proto3";
package predict;
option go_package = "/predict";

message CreatePredictRequest {
    int64 userID = 1;
    string sentence = 2;
    string intent = 3;
}

message CreatePredictResponse {
    bool success = 1;
    string message = 2;
}

message GetPredictsRequest {
    int64 userID = 1;
}

message GetPredictsResponse {
    bool success = 1;
    string message = 2;
    string data = 3;
}

message GetPredictRequest {
    int64 itemID = 1;
    int64 userID = 2;
}

message GetPredictResponse {
    bool success = 1;
    string message = 2;
    string data = 3;
}

message UpdatePredictRequest {
    int64 id = 1;
    string sentence = 2;
    string intent = 3;
}

message UpdatePredictResponse {
    bool success = 1;
    string message = 2;
}

message DeletePredictRequest {
    int64 id = 1;
}

message DeletePredictResponse {
    bool success = 1;
    string message = 2;
}

service PredictService {
    rpc CreatePredict(CreatePredictRequest) returns (CreatePredictResponse) {}
    rpc GetPredicts(GetPredictsRequest) returns (GetPredictsResponse) {}
    rpc GetPredict(GetPredictRequest) returns (GetPredictResponse) {}
    rpc UpdatePredict(UpdatePredictRequest) returns (UpdatePredictResponse) {}
    rpc DeletePredict(DeletePredictRequest) returns (DeletePredictResponse) {}
}